shell/components/ButtonMultiAction.vue

```diff
@@ -0,0 +1,41 @@
+<script setup lang="ts">
+import { computed, defineEmits } from 'vue';
+
+defineEmits(['click']);
+
+type Props = {
+  borderless?: boolean;
+  invisible?: boolean;
+}
+
+const props = defineProps<Props>();
+
+const buttonClass = computed(() => {
+  return {
+    borderless: props?.borderless,
+    invisible:  props?.invisible,
+  };
+});
```

Comment:
Sorry for asking, but would this not be more concise?
Not sure why `prop` may be undefined in your case as well, since it's already initialized with `defineProps`.

```suggestion
const props = defineProps<{
  borderless?: boolean | undefined;
  invisible?: boolean | undefined;
}>();

const buttonClass = computed(() => {
  return {
    borderless: props.borderless,
    invisible:  props.invisible,
  };
});
```

Reply:
It really just comes down to preference whether we want to move props into a separate type or not. I'm open to either approach.

Reply:
Yeah not a blocker as I said 🍻